### PR TITLE
Min nupcol

### PR DESCRIPTION
**Pull request by joakim-hove**
Add new OPM specific keyword `MINNPCOL` for the minimum value of `NUPCOL`.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
@tskille  / @totto82 are encouraged to comment on the default minimum `NUPCOL` value in the new `MINNPCOL` keyword.

**Comment by OPMUSER** [comment]
Since this is a OPM Flow keyword what about using MINUPCOL to keep with the eight character length, or NUPCOLMT. The latter will be next to the NUPCOL in the manual, so it will be easy to find.

I understand why you came up with MINNUPCOL but I think we should stick to eight character length; however, these are just suggestions, take them or leave them :-)

**Comment by joakim-hove** [comment]
> I understand why you came up with MINNUPCOL but I think we should stick to eight character length; however, these are just suggestions, take them or leave them :-)

Actually we have the eight character limit also for the flow specific keywords, so the keyword I came up with was `MINNPCOL`- i.e. without the `U` - the alternative `NUPCOLMT` is fine by me.

**Comment by joakim-hove** [comment]
No real comments to this - I'll merge it now. The name `MINNPCOL` or the default minimum value (6) can be changed later.

File: src/opm/parser/eclipse/EclipseState/Runspec.cpp [hunk 1]
```diff
@@ -457,6 +459,43 @@ bool SatFuncControls::operator==(const SatFuncControls& rhs) const
...
+    if (runspecSection.hasKeyword<ParserKeywords::MINNPCOL>()) {
+        const auto& min_item = runspecSection.getKeyword<ParserKeywords::MINNPCOL>().getRecord(0).getItem<ParserKeywords::MINNPCOL::VALUE>();
+        this->min_nupcol = min_item.get<int>(0);
+    }
+}
+
+void Nupcol::update(int value) {
+    if (value < this->min_nupcol)
+        OpmLog::note(fmt::format("OPM Flow uses {} as minimum NUPCOL value", this->min_nupcol));
+    this->nupcol_value = std::max(value, this->min_nupcol);
```
Comment (totto82) [inline src/opm/parser/eclipse/EclipseState/Runspec.cpp:480]: Is this a place where we should point the user to MINNPCOL keyword

**Comment by totto82** [comment]
I was actually looking at the PR right now. The default values needs some more testing, but as you say that can be changed later. I also think we should change the default of the original NUPCOL to 3.

File: src/opm/parser/eclipse/EclipseState/Runspec.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Runspec.cpp:480]: Good idea!

**Comment by joakim-hove** [comment]
> The default values needs some more testing, but as you say that can be changed later. I also think we should change the default of the original NUPCOL to 3.

I have no qualified opinion on this - just make a PR when you are ready.

File: src/opm/parser/eclipse/EclipseState/Runspec.cpp (hunk 1 above)
Comment (totto82) [reply src/opm/parser/eclipse/EclipseState/Runspec.cpp:480]: I can make a PR with some additional notes to the user + changing the default.

**Pull request merged**